src/ripple/overlay/impl/PeerImp.h

```diff
@@ -127,8 +128,8 @@ class PeerImp
     LedgerIndex maxLedger_ = 0;
     uint256 closedLedgerHash_;
     uint256 previousLedgerHash_;
-    std::list<uint256> recentLedgers_;
-    std::list<uint256> recentTxSets_;
+    std::deque<uint256> recentLedgers_;
```

Comment:
I put the policy I used for this PR in the PR description:

`deque` was chosen for the use cases that were only inserting and erasing at the front or back.
`vector` was chosen for the use cases that were only inserting and erasing at the back.